src/esphomelib/application.h

```diff
@@ -663,7 +665,13 @@ class Application {
   MakeESP32HallSensor make_esp32_hall_sensor(const std::string &name, uint32_t update_interval = 15000);
 #endif
 
+#ifdef USE_RTC_COMPONENT
+time::RTCComponent *make_rtc_component(const std::string &tz);
+#endif
 
+#ifdef USE_SNTP_COMPONENT
+time::SNTPComponent *make_sntp_component();
```

Comment:
Why not have the constructor pass in the time servers? Then we could a) set defaults and b) make sure the component is in a valid state when it's used (so that nobody can try to use this component without any servers).

```cpp
time::SNTPComponent *make_sntp_component(std::string server_1 = "1.pool.ntp.org", std::string server_2 = "2.pool.ntp.org", std::string server_3 = "3.pool.ntp.org");
```